aioredis/cluster/cluster.py

```diff
@@ -329,40 +303,52 @@ def masters_count(self):
     def slave_count(self):
         return self._cluster_manager.slaves_count
 
-    def get_nodes_entities(self):
-        return [node.address for node in self._cluster_manager.masters]
+    def get_nodes_entities(self, slaves=False):
+        slave_nodes = []
+        if slaves:
+            slave_nodes = [node.address for node in self.slave_nodes]
+        return [node.address for node in self.master_nodes] + slave_nodes
 
-    @asyncio.coroutine
-    def _get_raw_cluster_info_from_node(self, node):
-        conn = yield from create_redis(
+    @property
+    def master_nodes(self):
+        return self._cluster_manager.masters
+
+    @property
+    def slave_nodes(self):
+        return self._cluster_manager.slaves
+
+    async def _get_raw_cluster_info_from_node(self, node):
+        conn = await create_redis(
             node,
             db=self._db,
             password=self._password,
             encoding=self._encoding,
```

Comment:
Wouldn't it be easier to always use utf-8 for this internal connection? Then we could remove all those calls to decode in the constructor of ClusterNode. Also, the amount of places where it is not clear, whether we are dealing with strings or bytes would be somewhat reduced.

Reply:
Yes, I think you are right